File: java/WebAutomation-Series/src/test/java/kendogrid/simple/KendoGridTests.java
/*
 * Copyright 2020 Automate The Planet Ltd.
 * Author: Anton Angelov
 * Licensed under the Apache License, Version 2.0 (the "License");
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package kendogrid.simple;

import io.github.bonigarcia.wdm.WebDriverManager;
import kendogrid.components.*;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.*;

import java.util.List;
import java.util.concurrent.TimeUnit;

public class KendoGridTests {
    private WebDriver driver;
    private WebDriverWait wait;
    private KendoGrid kendoGrid;
    private final String OrderIdColumnName = "OrderID";
    private final String ShipNameColumnName = "ShipName";

    @BeforeClass
    public void classInit() {
        WebDriverManager.chromedriver().setup();
    }

    @BeforeTest
    public void testSetup() {
        driver = new ChromeDriver();
        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
        driver.manage().window().maximize();
        wait = new WebDriverWait(driver, 30);

        driver.navigate().to("https://demos.telerik.com/kendo-ui/grid/basic-usage");
        var consentButton = driver.findElement(By.id("onetrust-accept-btn-handler"));
        consentButton.click();

        kendoGrid = new KendoGrid(driver, driver.findElement(By.id("grid")));
    }

    @AfterTest
    public void afterClass() {
        driver.quit();
    }

    @Test
    public void filterContactName() throws Exception {
        kendoGrid.filter("ContactName", FilterOperator.CONTAINS, "Thomas");
        List<GridItem> items = kendoGrid.getItems();

        Assert.assertEquals(items.stream().count(), 1);
    }

    @Test
    public void sortContactTitleDesc() {
        kendoGrid.sort("ContactTitle", SortType.DESC);
        List<GridItem> items = kendoGrid.getItems();

        Assert.assertEquals(items.get(0).getContactTitle(), "Sales Representative");
        Assert.assertEquals(items.get(1).getContactTitle(), "Sales Representative");
    }

    @Test
    public void testCurrentPage() {
        var pageNumber = kendoGrid.getCurrentPageNumber();

        Assert.assertEquals(pageNumber, 1);
    }

    @Test
    public void getPageSize() {
        var pageNumber = kendoGrid.getPageSize();

        Assert.assertEquals(pageNumber, 20);
    }

     @Test
    public void getAllItems() {
         List<GridItem> items = kendoGrid.getItems();

        Assert.assertEquals(items.stream().count(), 20);
    }
}

File: dotnet/WebAutomation-Series/AutomateTelerikKendoGridWebDriverJavaScript/KendoGridTests.cs
// <copyright file="KendoGridTests.cs" company="Automate The Planet Ltd.">
// Copyright 2021 Automate The Planet Ltd.
// Licensed under the Apache License, Version 2.0 (the "License");
// You may not use this file except in compliance with the License.
// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// </copyright>
// <author>Anton Angelov</author>
// <site>https://automatetheplanet.com/</site>
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Firefox;
using OpenQA.Selenium.Support.UI;
using System;
using Ui.Automation.Core.Controls.Controls;
using Ui.Automation.Core.Controls.Enums;
using WebDriverManager;
using WebDriverManager.DriverConfigs.Impl;

namespace AutomateTelerikKendoGridWebDriverJavaScript
{
    [TestClass]
    public class KendoGridTests
    {
        private IWebDriver _driver;
        private WebDriverWait _wait;

        [ClassInitialize]
        public static void ClassInit(TestContext testContext)
        {
            new DriverManager().SetUpDriver(new ChromeConfig());
        }

        [TestInitialize]
        public void SetupTest()
        {
            _driver = new ChromeDriver();
            _driver.Manage().Timeouts().PageLoad = TimeSpan.FromSeconds(30);
            _wait = new WebDriverWait(_driver, TimeSpan.FromSeconds(30));

            _driver.Navigate().GoToUrl("https://demos.telerik.com/kendo-ui/grid/basic-usage");
            var consentButton = _driver.FindElement(By.Id("onetrust-accept-btn-handler"));
            consentButton.Click();
        }

        [TestCleanup]
        public void TeardownTest()
        {
            _driver.Quit();
        }

        [TestMethod]
        public void FilterContactName()
        {
            var kendoGrid = new KendoGrid(_driver, _driver.FindElement(By.Id("grid")));
            kendoGrid.Filter("ContactName", FilterOperator.Contains, "Thomas");
            var items = kendoGrid.GetItems<GridItem>();

            Assert.AreEqual(1, items.Count);
        }

        [TestMethod]
        public void SortContactTitleDesc()
        {
            var kendoGrid = new KendoGrid(_driver, _driver.FindElement(By.Id("grid")));
            kendoGrid.Sort("ContactTitle", SortType.Desc);
            var items = kendoGrid.GetItems<GridItem>();

            Assert.AreEqual("Sales Representative", items[0].ContactTitle);
            Assert.AreEqual("Sales Representative", items[1].ContactTitle);
        }

        [TestMethod]
        public void TestCurrentPage()
        {
            var kendoGrid = new KendoGrid(_driver, _driver.FindElement(By.Id("grid")));
            var pageNumber = kendoGrid.GetCurrentPageNumber();

            Assert.AreEqual(1, pageNumber);
        }

        [TestMethod]
        public void GetPageSize()
        {
            var kendoGrid = new KendoGrid(_driver, _driver.FindElement(By.Id("grid")));
            var pageNumber = kendoGrid.GetPageSize();

            Assert.AreEqual(20, pageNumber);
        }

        [TestMethod]
        public void GetAllItems()
        {
            _wait.Until(ExpectedConditions.ElementExists(By.Id("grid")));
            var kendoGrid = new KendoGrid(_driver, _driver.FindElement(By.Id("grid")));
            var items = kendoGrid.GetItems<GridItem>();

            Assert.AreEqual(20, items.Count);
        }
    }
}